js/ui/appDisplay.js

```diff
@@ -536,6 +545,13 @@ const AllView = new Lang.Class({
             Main.queueDeferredWork(this._redisplayWorkId);
         }));
 
+        this._addedFolderId = null;
+        IconGridLayout.layout.connect('folder-added',
```

Comment:
Could you pass a callback to `addFolder()` directly instead of adding the signal indirection and the extra state?